File: parralel-prog/Lab3/F1.cs
using System;
using System.Threading;
using static lab3.Lab3;

namespace lab3
{
    public class Function1
    { 
        public void Run()
        {
            //F1  1.2 C = A + B*(MO*ME) 
           lock (Lab3.locker)
           {
                //Thread.Sleep(1000);
                Vector A = new Vector(N);
                Vector B = new Vector(N);
                Matrix MO = new Matrix(N);
                Matrix ME = new Matrix(N);
               
                Vector result;

                A.FillVectorWithNumber(1);
                B.FillVectorWithNumber(1);
                MO.FillMatrixWithNumber(1);
                ME.FillMatrixWithNumber(1);

                Console.WriteLine("Function 1 start");
                result = A.sum(MO.multiply(ME).multiply(B));
                Console.WriteLine("Function 1 result \n" + result.toString());
                Console.WriteLine("Function 1 end \n");
           }
        }
    }

}

File: parralel-prog/Lab6/F1.cpp
#include "F1.h"

F1::F1(int N) {
	this->N = N;
}

Vector* F1::getResult() {
	return result;
}

DWORD WINAPI F1::startThread(void* param) {
	F1* This = (F1*)param;
	return This->run();
}
//1.12  A = B + C + D*(MD*ME) 
DWORD F1::run() {
	cout << "Task 1 start\n";
	Vector *D = new Vector(N);
	Vector *B = new Vector(N);
	Vector *C = new Vector(N);
	Matrix *MD = new Matrix(N);
	Matrix *ME = new Matrix(N);
	result = ((MD->multiply(ME))->multiply(D)->sum(B))->sum(C);
	cout << result->toString() << endl;
	cout << "Task 1 end\n";
	delete D;
	delete B;
	delete C;
	delete MD;
	delete ME;
	return 0;
}